packages/api/server/http.mts

```diff
@@ -515,6 +517,51 @@ router.post('/apps/:id/directories', cors(), async (req, res) => {
   }
 });
 
+router.options('/apps/:id/directories', cors());
+router.delete('/apps/:id/directories', cors(), async (req, res) => {
+  const { id } = req.params;
+
+  // TODO: validate and ensure path is not absolute
```

Comment:
I've wanted some path cleaning utils also in other places, this is a good idea. It's particularly useful given that LLMs are giving us paths which i have less confidence in.